lib/entity/fields/entity-date-field.ts

```diff
@@ -57,7 +57,7 @@ class EntityDateField extends EntityField {
   }
 
   private get valueAsNumber(): number {
-    return (this.value as Date).getTime();
+    return (this.value as Date).getTime() / 1000;
```

Comment:
I'm not sure what RediSearch will do it gets a fractional value here. I also don't trust JavaScript with floats that much. We should stick to integers. Should be we round it for floor it?

Reply:
It seems to search and index things OK, I couldn't find any definitive description so I assumed the index part is simply treating it as a number (which it's defined as in the index) and search + sort appears to handle the decimal part, so I was thinking the decimal would provide the ability to store things without much real loss of precision.

Whichever way it was rounded _might_ impact some things, so that might be something else to document or provide some control over.

Reply:
See what you think - I think this should allow people to use non-fractional times (rounded to the second) or have the extra precision if it's required and defined in the date string.